### ISSUE TITLE
Batch index or batch label using data generator

### ISSUE THREAD

**Issue by adiel2012** [keras-team/keras#14187]
Hello
I am using a data generator My_Custom_Generator  to generate batches. I am also creating a layer that needs the values of the labels for the batch that is being processed. How can I obtain the labels that I generate in the  __getitem__  method of the generator? The layer that I am creating is ArcFace. There are very good examples in the internet but they do not split the training set into batches

[code here](https://github.com/4uiiurz1/keras-arcface/blob/master/metrics.py)

```
class My_Custom_Generator(tf.keras.utils.Sequence) :
  ......
```

**Comment by sushreebarsa** [comment]
@adiel2012 Could you please refer to this[ link](https://www.tensorflow.org/api_docs/python/tf/keras/utils/Sequence#__getitem__) and the tested [gist](https://colab.research.google.com/gist/sushreebarsa/d1cd0d4c99423c3d8c820781b87adf1b/14187.ipynb) on colab of the provided code? 
Please let us know if it helps?
Thanks!
